### PR TITLE
Positive test scenarios for Organization

### PR DESCRIPTION
**Pull request by shweta83**
Few tests in tests/foreman/ui/test_organization.py needs refactor in airgun because of ui changes all tests won't pass in this PR.
pytest tests/foreman/ui/test_organization.py 
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope: None - storage: file
rootdir: /home/shwsingh/SatelliteQE/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, reportportal-5.0.8, cov-2.12.1, ibutsu-1.16, xdist-2.4.0, mock-3.6.1
collected 2 items                                                                                                                                                                                                 

tests/foreman/ui/test_organization.py ..                                                                                                                                                                    [100%]

================================================================================================ warnings summary ====================

-- Docs: https://docs.pytest.org/en/stable/warnings.html
==================================================================================== 2 passed, 7 warnings in 603.20s (0:10:03) ====================================================================================

### CODE REVIEW

File: tests/foreman/ui/test_organization.py [hunk 1]
```diff
@@ -289,3 +322,83 @@ def test_positive_download_debug_cert_after_refresh(session):
...
+    with session:
+        assert (
+            session.errata.search(CUSTOM_REPO_ERRATA_ID, applicable=False)[0]['Errata ID']
+            == CUSTOM_REPO_ERRATA_ID
+        )
+        session.organization.select(org_name="Default Organization")
+        assert not session.errata.search(CUSTOM_REPO_ERRATA_ID, applicable=False)
+
+        @request.addfinalizer
+        def _cleanup():
```
Comment (jameerpathan111) [inline tests/foreman/ui/test_organization.py:355]: @shweta83 I think we should move this cleanup part in `module_repos_col` fixture itself.

File: tests/foreman/ui/test_organization.py [hunk 2]
```diff
@@ -16,16 +16,49 @@
...
+
+CUSTOM_REPO_URL = settings.repos.yum_6.url
+CUSTOM_REPO_ERRATA_ID = settings.repos.yum_6.errata[2]
+logger = logging.getLogger('robottelo')
+
+
+@pytest.fixture(scope='module')
+def module_repos_col(module_org, module_lce, default_sat, request):
+    default_sat.api.Parameter(
+        name='remote_execution_connect_by_ip', value='Yes', organization=module_org.id
```
Comment (jameerpathan111) [inline tests/foreman/ui/test_organization.py:47]: @shweta83 can you explain why we need to update this setting for tests?

File: tests/foreman/ui/test_organization.py (hunk 1 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py]: Agree. Moved this to fixture.

File: tests/foreman/ui/test_organization.py (hunk 2 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py]: This was added for different test which is not yet added. Updated this.

**Review by jameerpathan111 — APPROVED** [review]

File: tests/foreman/ui/test_organization.py [hunk 3]
```diff
@@ -16,16 +16,56 @@
...
+            SatelliteToolsRepository(),
+            # As Satellite Tools may be added as custom repo and to have a "Fully entitled" host,
+            # force the host to consume an RH product with adding a cdn repo.
+            RHELAnsibleEngineRepository(cdn=True),
+            YumRepository(url=CUSTOM_REPO_URL),
+        ],
+    )
+    repos_collection.setup_content(module_org.id, module_lce.id)
+
+    @request.addfinalizer
```
Comment (mshriver) [inline tests/foreman/ui/test_organization.py:59]: This pattern is quite redundant within a fixture function.

Just move the try/except block to after a `yield` where currently the function `return`s

https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended

File: tests/foreman/ui/test_organization.py [hunk 4]
```diff
@@ -289,3 +329,74 @@ def test_positive_download_debug_cert_after_refresh(session):
...
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    product_name = gen_string('alpha')
+    product_label = gen_string('alpha')
+    product_description = gen_string('alpha')
+    gpg_key = entities.GPGKey(
+        content=read_data_file(VALID_GPG_KEY_FILE), organization=module_org
+    ).create()
```
Comment (mshriver) [inline tests/foreman/ui/test_organization.py:387]: This all seems like something that should be done during test setup phase, in a fixture, as the test case is trying to provide coverage for the UI behavior when switching organization selection.

This also is a test that should fall under RBAC, are you sure we don't have coverage for it there?

File: tests/foreman/ui/test_organization.py [hunk 5]
```diff
@@ -289,3 +329,74 @@ def test_positive_download_debug_cert_after_refresh(session):
...
+            session.errata.search(CUSTOM_REPO_ERRATA_ID, applicable=False)[0]['Errata ID']
+            == CUSTOM_REPO_ERRATA_ID
+        )
+        session.organization.select(org_name="Default Organization")
+        assert not session.errata.search(CUSTOM_REPO_ERRATA_ID, applicable=False)
+
+
+@pytest.mark.tier2
+@pytest.mark.skipif((not settings.robottelo.REPOS_HOSTING_URL), reason='Missing repos_hosting_url')
+def test_negative_product_view_organization_switch(session, module_org):
```
Comment (mshriver) [inline tests/foreman/ui/test_organization.py:364]: I don't think I'd consider this a negative test scenario.  There's no error/warning/exception/failure condition that you're hitting here.  Its a positive test that RBAC for products works at the organization selection scope.

File: tests/foreman/ui/test_organization.py [hunk 6]
```diff
@@ -289,3 +329,74 @@ def test_positive_download_debug_cert_after_refresh(session):
...
+    :Steps:
+            1. Create an Organization having a product and verify that product is present in
+               the Organization.
+            2. Switch the Organization to default and verify that product is not visible in it.
+
+
+    :expectedresults: Verify that the Product belonging to one Organization is not visible in
+                      another organization.
+
+    :CaseLevel: Integration
```
Comment (mshriver) [inline tests/foreman/ui/test_organization.py:378]: Is there a BZ associated with this test coverage?

Also, for metadata, if the test case *is* negative (see other comment), it needs a testimony token to set that metadata:
https://github.com/SatelliteQE/robottelo/blob/master/testimony.yaml#L130

The default is positive.

File: tests/foreman/ui/test_organization.py [hunk 7]
```diff
@@ -16,16 +16,56 @@
...
 from robottelo.manifests import original_manifest
 from robottelo.manifests import upload_manifest_locked
+from robottelo.products import RepositoryCollection
+from robottelo.products import RHELAnsibleEngineRepository
+from robottelo.products import SatelliteToolsRepository
+from robottelo.products import YumRepository
+
+CUSTOM_REPO_URL = settings.repos.yum_6.url
+CUSTOM_REPO_ERRATA_ID = settings.repos.yum_6.errata[2]
+logger = logging.getLogger('robottelo')
```
Comment (mshriver) [inline tests/foreman/ui/test_organization.py:41]: This should not be necessary or used. The logger instance should be imported: `from robottelo.logging import logger`

File: tests/foreman/ui/test_organization.py (hunk 3 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py:56]: Updated as per comments.

File: tests/foreman/ui/test_organization.py (hunk 4 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py]: Updated.

File: tests/foreman/ui/test_organization.py (hunk 5 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py]: Updated it s positive. RBAC doesn't have this test scenario.

File: tests/foreman/ui/test_organization.py (hunk 6 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py:363]: Updated test as positive. It doesn't have any BZ associated.

File: tests/foreman/ui/test_organization.py (hunk 7 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py]: Updated.

File: tests/foreman/ui/test_organization.py [hunk 8]
```diff
@@ -289,3 +324,46 @@ def test_positive_download_debug_cert_after_refresh(session):
...
     finally:
         entities.Subscription(organization=org).delete_manifest(data={'organization_id': org.id})
+
+
+@pytest.mark.tier2
+def test_positive_errata_view_organization_switch(
+    session, module_org, module_lce, module_repos_col, default_sat
+):
+    """Verify no errata list visible on Organization switch
+    :id: a5317b2d-2593-4bee-9b20-78066cf2964c
```
Comment (latran) [inline tests/foreman/ui/test_organization.py:334]: Have a new line for each of the docstring so it can be consistent with the entire file.

File: tests/foreman/ui/test_organization.py [hunk 9]
```diff
@@ -289,3 +324,46 @@ def test_positive_download_debug_cert_after_refresh(session):
...
+        )
+        session.organization.select(org_name="Default Organization")
+        assert not session.errata.search(CUSTOM_REPO_ERRATA_ID, applicable=False)
+
+
+@pytest.mark.tier2
+@pytest.mark.skipif((not settings.robottelo.REPOS_HOSTING_URL), reason='Missing repos_hosting_url')
+def test_positive_product_view_organization_switch(session, module_org, module_product):
+    """Verify product created in one organization is not visible in another
+    :id: 47f3580e-6071-4499-bf27-1ded62a7d3e3
```
Comment (latran) [inline tests/foreman/ui/test_organization.py:356]: Same here with the docstring.

**Review by latran — CHANGES_REQUESTED** [review]

File: tests/foreman/ui/test_organization.py (hunk 9 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py:362]: Updated.

File: tests/foreman/ui/test_organization.py (hunk 8 above)
Comment (shweta83) [reply tests/foreman/ui/test_organization.py:335]: Updated.

**Comment by latran** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py

**Comment by jyejare** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py

**Comment by jyejare** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_organization.py

**Comment by jyejare** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_organization.py

**Comment by jyejare** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_organization.py

**Comment by jyejare** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py

**Comment by jyejare** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py

**Comment by JacobCallahan** [comment]
@shweta83 we're still seeing failures in the PRT. If these aren't something you can address within a day, does this PR still need to be in before 6.10 branching?

**Comment by jyejare** [comment]
@JacobCallahan she has already confirmed on the chat that it can later be cherrypicked back to 6.10 and hence its a non-blocking PR for branching.

**Comment by shweta83** [comment]
> @shweta83 we're still seeing failures in the PRT. If these aren't something you can address within a day, does this PR still need to be in before 6.10 branching?

@JacobCallahan We are good to branch 6.10. It's a non-blocking PR.

**Comment by jyejare** [comment]
@shweta83 Did you look into 2 test failures of PRT already? and are we good to merge it?

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py

**Comment by latran** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py -k test_positive_errata_view_organization_switch

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py -k test_positive_product_view_organization_switch

**Review by mshriver — APPROVED** [review]

**Comment by mshriver** [comment]
@shweta83 please rebase so there is no merge commit, and re-run the pull request tester. Let's get this merged! :)

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py -k test_positive_errata_view_organization_switch

**Comment by mshriver** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py -k test_positive_errata_view_organization_switch

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_organization.py -k test_positive_errata_view_organization_switch
